Cargo.toml

```diff
@@ -18,31 +18,30 @@ coveralls = { repository = "softprops/shipflit" }
 maintenance = { status = "actively-developed" }
 
 [dependencies]
-base64 = "0.10"
-byteorder = "1.3.1"
-bytes = "0.4"
-chrono = { version = "0.4", optional = true, features = ["serde"] }
-flate2 = "1.0"
-futures = "0.1"
-http = "0.1"
-hyper = "0.12"
-hyper-openssl = { version = "0.7", optional = true }
-hyperlocal = { version = "0.6", optional = true }
-log = "0.4.6"
-mime = "0.3.13"
-openssl = { version = "0.10", optional = true, features = ["vendored"] }
-serde = { version = "1.0", features = ["derive"] }
-serde_json = "1.0"
-tar = "0.4"
-tokio = "0.1"
-tokio-codec = "0.1"
-tokio-io = "0.1"
-url = "1.7"
+base64 = "0.11.0"
+bytes = "0.4.12"
+chrono = { version = "0.4.9", optional = true, features = ["serde"] }
+flate2 = "1.0.13"
+futures = "0.3.1"
+http = "0.1.19"
+hyper = "0.13.0-alpha.4"
+hyper-openssl = { version = "0.8.0-alpha.4", optional = true }
+#hyperlocal = { version = "0.6", optional = true }
+log = "0.4.8"
+mime = "0.3.14"
+openssl = { version = "0.10.25", optional = true, features = ["vendored"] }
+serde = { version = "1.0.102", features = ["derive"] }
```

Comment:
The minor for versions is not needed: https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html#caret-requirements

Reply:
Do you mean the patch version?

I use an automated tool for bumping dependencies (cargo upgrade). It adds the full semver to the dependencies.

- The major version is required because it specifies API version
- the minor version is required because it *may* specify minimum feature set
- if we know that patch n exists, why allow the use of patch <n given that it's functionally identical, but may have performance or logic bugs?

Is there a downside to including the full version?

I've not been able to find any prior art on best practice here.

Reply:
See this [issue](https://github.com/killercup/cargo-edit/issues/126)